pkg/keys/sql.go

```diff
@@ -150,16 +151,26 @@ func MakeSQLCodec(tenID roachpb.TenantID) SQLCodec {
 	}
 }
 
-// SystemSQLCodec is a SQL key codec for the system tenant.
+// SystemSQLCodec is a SQL key codec for the system tenant with tenantID {2}.
+var SystemSQLCodec = func() SQLCodec {
+	if roachpb.EnableExperimentalUA {
+		return MakeSQLCodec(roachpb.TenantTwo)
+	}
+	return PrefixlessSystemSQLCodec
+}()
+
+// PrefixlessSystemCodec is a SQL codec for the system tenant with tenantID {1}.
 //
 // NB: We don't use MakeSQLCodec here since the system tenant is special and its
 // prefix is empty, rather than the start of its span, so the Codec for it wants
 // the empty key, rather than start key, as its buf. Ideally we would set the
 // endKey to TableDataMax instead of MaxKey here, but TableDataMax is currently
 // defined in terms of this codec. We would want to fix this if/when we make the
 // tenant with ID one non-system, but we'll get rid of/rename this then as well.
-var SystemSQLCodec = SQLCodec{
-	sqlEncoder: sqlEncoder{&MinKey, &MaxKey, roachpb.SystemTenantID},
+//
+// TODO(chandrat) revisit minKey and maxKeys.
+var PrefixlessSystemSQLCodec = SQLCodec{
```

Comment:
I think it would be more accurate if we named this:

```suggestion
var PrefixlessSQLCodec = SQLCodec{
```